pandas/core/reshape/merge.py

```
@@ -1954,42 +1954,65 @@ def _validate_left_right_on(self, left_on, right_on):
             if not left_unique:
+                sample = sample_duplicates(left_keys)
                 raise MergeError(
                     "Merge keys are not unique in left dataset; not a one-to-one merge"
+                    f"Offending keys (sample): {sample}"
                 )
             if not right_unique:
+                sample = sample_duplicates(right_keys)
                 raise MergeError(
                     "Merge keys are not unique in right dataset; not a one-to-one merge"
+                    f"Offending keys (sample): {sample}"
                 )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)